### PR TITLE
refactor: remove github.com/pkg/errors dependency

### PR DESCRIPTION
**Pull request by catthehacker**
- moved git from common in it's own package
- improved error messages
- Container.Close() returns an error if closing the docker socket fails
- Container.Exec() returns a decriptive error for 127 file not found

_Originally posted by @ChristopherHX in https://github.com/nektos/act/pull/1077#pullrequestreview-962491683_

- forbid `github.com/pkg/errors` via `golangci-lint`

### CODE REVIEW

**Review by ChristopherHX — APPROVED** [review]
After I noticed that the removed depenency of this PR is abandoned, I fully agree with removing it.
Change summary, not mentioned in text form:
- moved git from common in it's own package
- improved error messages
- Container.Close() returns an error if closing the docker socket fails
- Container.Exec() returns a decriptive error for 127 file not found

File: pkg/container/docker_run.go [hunk 1]
```diff
@@ -276,8 +275,11 @@ func (cr *containerReference) connect() common.Executor {
 func (cr *containerReference) Close() common.Executor {
 	return func(ctx context.Context) error {
 		if cr.cli != nil {
-			cr.cli.Close()
+			err := cr.cli.Close()
 			cr.cli = nil
+			if err != nil {
+				return fmt.Errorf("failed to close client: %w", err)
```
Comment (KnisterPeter) [inline pkg/container/docker_run.go:281]: Maybe this is more likely a logged warning? Why fail completly in this case?

Also you do ignore the errors over here: https://github.com/nektos/act/pull/1077/files#diff-ac725a1756aff41010f7e72b3e3f0f63677898bd5eea991096438ffe9c45f9ec

File: pkg/container/docker_run.go [hunk 2]
```diff
@@ -579,14 +583,17 @@ func (cr *containerReference) exec(cmd []string, env map[string]string, user, wo
...
+		case 0:
 			return nil
+		case 127:
+			return fmt.Errorf("exitcode '%d': command not found, please refer to https://github.com/nektos/act/issues/107 for more information", inspectResp.ExitCode)
+		default:
+			return fmt.Errorf("exitcode '%d': failure", inspectResp.ExitCode)
 		}
-
-		return fmt.Errorf("exit with `FAILURE`: %v", inspectResp.ExitCode)
 	}
```
Comment (KnisterPeter) [inline pkg/container/docker_run.go:597]: 👍

**Review by KnisterPeter — COMMENTED** [review]
Look good overall.

**Review by KnisterPeter — DISMISSED** [review]

**Review by ChristopherHX — APPROVED** [review]
Reapprove...
Can we merge this soon? Everything is fine with this change.

**Review by cplee — APPROVED** [review]

**Pull request merged**